lib/restbase.js

```diff
@@ -307,6 +318,17 @@ RESTBase.prototype._request = function (req) {
             self.metrics.endTiming([statName + statusClass, statName + 'ALL'], startTime);
             throw err;
         });
+        if (match.permissions
+                && Array.isArray(match.permissions)
+                && match.permissions.length > 0) {
+            reqHandlerPromise = reqHandlerPromise.then(function(res) {
```

Comment:
@gwicke when I made it I had in mind saving some pointless requests, but now I'm thinking that making the worst case speed better is more important..